controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -976,9 +977,13 @@ public OperationContext createStreamContext(String scopeName, String streamName,
         Stream streamObj = getStream(scope, stream, context);
         return Futures.completeOn(streamObj.completeCommittingTransactions(record, context, writerTimes, writerIdToTxnOffsets), executor)
                 .thenAcceptAsync(result -> {
-                    streamObj.getNumberOfOngoingTransactions(context).thenAccept(count ->
-                            TransactionMetrics.reportOpenTransactions(scope, stream, count));
-                    TransactionMetrics.getInstance().commitTransactionComplete(timer.getElapsed());
+                    streamObj.getAllOrderedTxns(context)
```

Comment:
instead of doing it again here, whereby adding to the commit-txn latency, we could do it when this was called first -- trystartcommittingTxn and update the count at that point.

Reply:
fixed